### PR TITLE
Documentation (Assignment 2.8)

### LINKED ISSUE #21: Documentation(Assignment 2.8)
1. Use Sphinx to generate HTML documents for your library

2. Place the doc related source in the appropriate directory according to project layout.

3. Setup a build job for documentation that does the following:
  a. Docs build job
      i. Conda setup
      ii. Display the directory tree in the workflow before building docs.
      iii. Build docs using sphinx.
      iv. Display the directory tree in the workflow before building docs.
      v. Upload the built docs html using upload action.

### PR DESCRIPTION
**Pull request by Dhikshita-macherla**
Hence closes #21 
1. Used Sphinx to generate HTML documents for your library 
2. Placed the doc related source in the docs directory.
3. Created the workflow setup for a build job 
  a. Docs build job
  - Displayed the directory tree in the workflow before building docs.
  - Built docs using sphinx.
  - Displayed the directory tree in the workflow before building docs.
  - Uploaded the built docs html using upload action.
  
Screenshots
workflow
image

Jobs op
image

### CODE REVIEW

File: .github/workflows/documentation.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        with:
+          name: html_arti
+          path: docs/
+
+      - name: Download the docs
+        uses: actions/download-artifact@v4
+        with:
+          name: html_arti
+          path: docs/
+
```
Comment (abilashravi-ta) [inline .github/workflows/documentation.yml:58]: This download step is not needed.

File: .github/workflows/documentation.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      - name: Tree after build
+        run: |
+          tree
+
+      - name: Upload the build docs
+        uses: actions/upload-artifact@v4
+        with:
+          name: html_arti
+          path: docs/
```
Comment (abilashravi-ta) [inline .github/workflows/documentation.yml:51]: inside `docs/` there will be `_build/html/` just uploading that `html/` file alone is sufficient.

File: .github/workflows/documentation.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+          environment-file: deploy/conda/env.yml
+          auto-activate-base: false
+
+      - name: Install Tree Cli
+        run: |
+          sudo apt install tree
+
+      - name: Tree before build
+        run: |
+          tree
```
Comment (abilashravi-ta) [inline .github/workflows/documentation.yml:31]: Why do you still have that `main.py` file. That should have been remove during the _workflow scripts_ assignment itself.

image

**Review by abilashravi-ta — CHANGES_REQUESTED** [review]
All good.! 👏 

just fix the suggested changes alone.

File: .github/workflows/documentation.yml (hunk 3 above)
Comment (Dhikshita-macherla) [reply .github/workflows/documentation.yml:31]: I've corrected all the changes that you specified. Can you please take a look at it?

**Review by abilashravi-ta — APPROVED** [review]

**Pull request merged**